### ISSUE TITLE
Is there a way to open a deafult browser with the App.run() method?

### ISSUE THREAD

**Issue by ChandlerTayek** [rstudio/py-shiny#300]
Is there any way to directly open up a default browser when `App.run()` is called without relying on something like `webbrowser.open(url)`?

**Comment by jcheng5** [comment]
Do you mean, would it be possible for the `shiny` package to invoke `webbrowser.open` at the appropriate time with the appropriate URL, instead of you having to do it manually? Or are you trying to avoid `webbrowser` altogether?

**Comment by ChandlerTayek** [comment]
Yes to your first point. I'm currently calling webbrowser.open(URL) before initiating the App.run() function. I was thinking it might be nice to pass a boolean into the App constructor or the .run() function to have Shiny handle calling the web-browser instead to avoid any race conditions like the server taking a while to start and the open browser timing out cause it doesn't keep re-trying past like 4 secs or so.
